packages/snaps-utils/src/manifest/validation.ts

```diff
@@ -141,7 +143,62 @@ export const SnapIdsStruct = refine(
 
 export type SnapIds = Infer<typeof SnapIdsStruct>;
 
-export const ChainIdsStruct = array(ChainIdStruct);
+export const ChainIdsStruct = refine(
+  size(array(ChainIdStruct), 1, Infinity),
+  'Chain IDs',
+  (value) => {
+    if (Array.isArray(value) && value.every((val) => isChainId(val))) {
+      return true;
```

Comment:
Why do we need special validation logic for this? It looks like the struct already would validate this.

If we need it, it should be tested.

Reply:
I suppose I re-implemented the validation logic here because otherwise how would I know when to return an error. But moreover, those errors don't show in the tests and they're standard superstruct errors, so maybe I can remove this?

Reply:
I believe it checks the root type first. So adding identical validation logic as a refinement doesn't add anything as far as I can tell.